.github/workflows/benchmark.yaml

```diff
@@ -0,0 +1,132 @@
+name: Benchmark
+
+on:
+  pull_request:
+    branches: [master]
+    types: [labeled, synchronize]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    if: >-
+      (github.event.action == 'labeled' && github.event.label.name == 'performance') ||
+      (github.event.action == 'synchronize' && contains(github.event.pull_request.labels.*.name, 'performance'))
+
+    env:
+      MSG_FOOTER: |-
+        <br/>
+
+        Workflow: [${{ github.run_id }}](/${{ github.repository }}/actions/runs/${{ github.run_id }})
+        *Adding new commits will generate a new report*
+
+    steps:
+      - name: Post comment
+        uses: jungwinter/comment@v1
+        id: create_comment
+        with:
+          type: create
+          body: |
+            Started a benchmark for this pull request.
+            This comment will be updated with the results.
+            ${{ env.MSG_FOOTER }}
+          issue_number: ${{ github.event.number }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - uses: actions/checkout@v2
+        with:
+          repository: yewstack/js-framework-benchmark
+
+      - name: Configure benchmark
+        run: |
+          search="{ git = \"https://github.com/jstarry/yew\", branch = \"keyed-list\" }"
+          replace="{ git = \"${{ github.event.pull_request.head.repo.html_url }}\", branch = \"${{ github.event.pull_request.head.ref }}\" }"
+          input=$(cat frameworks/keyed/yew/Cargo.toml)
+          output=${input//"$search"/"$replace"}
+          if [[ "$input" == "$output" ]]; then
+            echo "ERROR: failed to configure Cargo.toml"
+            exit 1
+          fi
+          echo "$output" > frameworks/keyed/yew/Cargo.toml
+          echo "$output"
+
+      - name: Setup ChromeDriver
+        uses: nanasess/setup-chromedriver@master
+
+      - name: Setup Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+
+      - uses: actions-rs/install@v0.1
+        with:
+          crate: wasm-bindgen-cli
+          version: latest
+          use-tool-cache: true
+
+      - uses: actions-rs/install@v0.1
+        with:
+          crate: wasm-pack
+          version: latest
+          use-tool-cache: true
+
+      - uses: actions-rs/install@v0.1
+        with:
+          crate: https
+          version: latest
+          use-tool-cache: true
+
+      - name: Start Server
+        run: http -p 8080 &
+
+      - name: Setup Node
+        uses: actions/setup-node@v1
+        with:
+          node-version: 12
+
+      - name: npm Install
+        run: |
+          npm install
+          (cd webdriver-ts && npm install)
+          (cd webdriver-ts-results && npm install)
+
+      - name: Build
+        run: |
+          npm run build-prod
+          (cd webdriver-ts && npm run build-prod)
+
+      - name: Benchmark
+        run: npm run bench -- --headless
```

Comment:
I'm unsure about how reliable the benchmarks will be reliable/accurate if run on Github actions – are all containers in which workflows are run guaranteeed to have the same performance?

Reply:
Yeah, this is a good point but my hope was that this wouldn't be an issue due to the way we compare against the baseline. We aren't comparing run #1 against run #2. We are comparing 3 different frameworks from one single benchmarks run.

Reply:
Ah. I think Github Actions spins stuff up in containers, and it may be that even during a run the resources available to a container change to manage supply/demand amongst all jobs? I'll look at the documentation and see what they say.

Reply:
I don't think this is valid criticism of this PR since the current process already asks users to open a PR over at the `js-framework-benchmark` which also runs the benchmark on GitHub Actions.

You still make a good point, but I don't think it's that much of a problem. The workflow isn't running in a docker container, it's a virtual machine and if [the documentation](https://help.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners#supported-runners-and-hardware-resources) is to be trusted, these have consistent hardware resources.

Reply:
That's cool then!

Reply:
`Continuous Benchmark` has a paragraph about ["Stability of Virtual Environment"](https://github.com/marketplace/actions/continuous-benchmark#stability-of-virtual-environment) which says:

> As far as watching the benchmark results of examples in this repository, the amplitude of the benchmarks is about +- 10~20%.